File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robot/Wheels.java
package org.firstinspires.ftc.teamcode.robot;

import android.annotation.SuppressLint;
import com.qualcomm.robotcore.hardware.DcMotorEx;
import org.firstinspires.ftc.teamcode.system.BotSystem;
import org.jetbrains.annotations.NotNull;

import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;

public class Wheels extends HardwareMapElement {
	
	private static List<String> wheelNames = Arrays.asList("FrontLeft", "FrontRight", "BackLeft", "BackRight");
	public List<DcMotorEx> wheels = null;
	
	@SuppressLint("NewApi")
	@Override
	protected void moreInit(@NotNull BotSystem botSystem) {
		wheels = wheelNames.stream()
				         .map(s -> hardwareMap.get(DcMotorEx.class, s))
				         .collect(Collectors.toList());
	}
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robot/Wheels.kt
package org.firstinspires.ftc.teamcode.robot

import com.qualcomm.robotcore.hardware.DcMotorEx
import org.firstinspires.ftc.teamcode.system.BotSystem

class Wheels : HardwareMapElement() {
    var wheels: List<DcMotorEx>? = null
    override fun moreInit(botSystem: BotSystem) {
        wheels = wheelNames.map {
            hardwareMap.get(DcMotorEx::class.java, it)
        }.toList()
    }

    companion object {
        private val wheelNames =
            listOf("FrontLeft", "FrontRight", "BackLeft", "BackRight")
    }
}
